QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,199 @@
+This document is a declaration of software quality for the `iceoryx_hoofs`, `iceoryx_posh` and `iceoryx_binding_c` packages, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html). These packages are part of the Eclipse iceoryx middleware.
+
+# Quality Declaration
+
+The packages `iceoryx_hoofs`, `iceoryx_posh` and `iceoryx_binding_c` claims to be in the **Quality Level 2** category. In the following documentation these packages are referred as `iceoryx`. Some of the rationales in this document refer to the complete project.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Requirements for Quality Level 2 in REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+See [the Readme](https://github.com/eclipse-iceoryx/iceoryx#quality-levels--platforms) for a full list of the quality levels of iceoryx components and [Contributing Guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#quality-levels) for a description of the quality levels.
+
+## Version Policy [1]
+
+### Version Scheme [1.i]
+
+`iceoryx` follows the `semver` scheme for versioning.
+In particular the version numbers have the following meaning:
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+1. MAJOR version when you make incompatible API/ABI changes,
+2. MINOR version when you add functionality in a backwards compatible manner. This means that changes are API compatible and ideally ABI stable if not otherwise clearly stated in the release notes
```

Comment:
@elfenpiff We should stick as much as possible to the semver policy, especially for the iceoryx packages that are integrated into ROS 2. Generally the packages are as shared library present in ROS 2 but we need to be especially careful when working on releases because we need also to ensure that we are "shared memory compatible". 

Here some rules from the [ROS 2 Developer Guide]() regarding the ABI stability:

> - For compiled code, the ABI is considered part of the public interface. Any change that requires recompiling dependent code is considered major (breaking).
     - ABI breaking changes can be made in a minor version bump before a distribution release (getting added to the rolling release).

The exception where it is allowed is the rolling distribution where the development in ROS 2 is ongoing. We can break the API but only under the assumption that also the corresponding dependencies (in our case cyclonedds or maybe rmw_cyclonedds_cpp) get updated too. This need some coordination though.

That's why we need to have this strict policy.